bpdm-gate-api/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/api/GateBusinessPartnerApi.kt

```diff
@@ -95,4 +93,22 @@ interface GateBusinessPartnerApi {
         @ParameterObject @Valid paginationRequest: PaginationRequest = PaginationRequest()
     ): PageDto<BusinessPartnerOutputDto>
 
+    @Operation(
+        summary = "Create or update business partners from uploaded CSV file",
+        description = "Create or update generic business partners. " +
+                "Updates instead of creating a new business partner if an already existing external ID is used. " +
+                "The same external ID may not occur more than once in a single request. " +
+                "For file upload request, the maximum number of business partners in file limited to \${bpdm.api.upsert-limit} entries.",
+    )
+    @ApiResponses(
+        value = [
+            ApiResponse(responseCode = "200", description = "Business partners were successfully updated or created"),
+            ApiResponse(responseCode = "400", description = "On malformed legal entity request", content = [Content()]),
+        ]
+    )
+    @PostMapping("/input/business-partners/uploadCsv", consumes = ["multipart/form-data"])
```

Comment:
REST Conformity: From this release on, I propose to get our REST API definitions in order and more closely follow the REST principles. In this spirit, paths should lead to resources and these resources should be named as proper nouns. So I would suggest the following changes:

- Find a proper noun resource name for the CSV-Upload resource
- give it it's own path independent of "business-partners". Either keep it under "input" path or even put the resource in the root path (both solutions have arguments for and against, I would be OK with both)

Reply:
With this comment, 
Sure, I will use proper noun resource name for the CSV-upload.
But just wanted to confirm, are you suggesting me to create different api client and controllers as well? I covered it under gate business partner api as ultimately business partner data need to be uploaded.

Reply:
It is arguable if it ultimately really just business partner that is uploaded. From a REST perspective you create a new upload process resource. Assuming the resource would be called "partner-upload-process" (just an example) then I would see two possible paths where this resource is available:

input/partner-upload-process
or
partner-upload-process

Regarding controller: I would suggest to create a new controller for that resource. It will make endpoint annotations pretty straight forward and it establishes more clearly that this upload is its own resource.

Reply:
I agree it is arguable, what I interpreted from requirement currently is just business partner that is going to uploaded. 
May be once feature is ready then we can come to concrete solution, but regarding resource noun and controller I can find feasibility and keep it ready on this branch at least.